kernel/src/lib.rs

```diff
@@ -70,9 +70,8 @@ pub mod net;
 pub mod prelude;
 mod process;
 mod sched;
-pub mod softirq_id;
+pub mod softirq;
```

Comment:
Bottom-half mechanisms should be useful to all drivers, which may exists as separate crates. As such, bottom-half mechanism should not be extracted out of `ostd` into `kernel`, but as a component. So I think we should have a `softirq` crate.